modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -339,29 +333,28 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
     declareMeshProperty("max_radius_meta",
                         getMeshProperty<Real>("max_radius_meta", name() + "_2D"));
 
-  // id swap info after extrusion
-  std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>> id_map{
+  // Store pin region ids and block names for id swap after extrusion if needed
+  // by future mesh generators
+  std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>> region_id_map{
       {_pin_type, _region_ids}};
-  declareMeshProperty("pin_region_ids", id_map);
-
-  // transferring reactor parameters and PolygonConcentricCircleMeshGenerator parameters to pin mesh
-  declareMeshProperty("reactor_params_name", std::string(_reactor_params));
+  declareMeshProperty("pin_region_ids", region_id_map);
+  std::map<subdomain_id_type, std::vector<std::vector<std::string>>> block_name_map;
+  block_name_map[_pin_type] = _block_names;
```

Comment:
how is this initialized if the user does not set it? seems to me like _block_names is an empty vector then?

Reply:
Correct, this will be an empty vector. Later on, AMG will check whether this vector is empty to figure out whether block names were provided by the user or not